### PR TITLE
Refactor Code-Mode to use CodeEditor component

### PR DESCRIPTION
**Pull request by habdelra**
This splits out a `CodeEditor` component from the code sub-mode to manage the source code editing specifically

### CODE REVIEW

**Review by jurgenwerk — APPROVED** [review]

File: packages/host/app/components/operator-mode/code-editor.gts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    }
+    throw new Error(
+      `cannot access file contents ${this.codePath} before file is open`,
+    );
+  }
+
+  @action
+  private initializeMonacoCursorPosition() {
+    if (this.args.selectedDeclaration?.path?.node.loc) {
+      let { start } = this.args.selectedDeclaration.path.node.loc;
```
Comment (lukemelia) [inline packages/host/app/components/operator-mode/code-editor.gts]:
```suggestion
    let loc =  this.args.selectedDeclaration?.path?.node.loc;
    if (loc) {
      let { start } = loc;
```

File: packages/host/app/components/operator-mode/code-editor.gts [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    declaration: ModuleDeclaration,
+  ) {
+    if (declaration.path?.node.loc) {
+      let { start, end } = declaration.path?.node.loc;
+      let currentCursorPosition = this.monacoService.getCursorPosition();
+      if (
+        currentCursorPosition &&
+        (currentCursorPosition.lineNumber < start.line ||
+          currentCursorPosition.lineNumber > end.line)
+      ) {
```
Comment (lukemelia) [inline packages/host/app/components/operator-mode/code-editor.gts:134]: Does this mean that if there is no cursor position, then the declaration does get the cursor position moved to it? Or is there always a cursor position?

**Review by lukemelia — APPROVED** [review]
Nice work!

File: packages/host/app/components/operator-mode/code-editor.gts (hunk 2 above)
Comment (habdelra) [reply packages/host/app/components/operator-mode/code-editor.gts:134]: i think @FadhlanR can answer this question better. i just moved it to a new place...

**Review by burieberry — APPROVED** [review]

**Pull request merged**